tensorflow.go

```diff
@@ -0,0 +1,172 @@
+package tensorflow
+
+import (
+	"encoding/binary"
+	"fmt"
+	"reflect"
+	"unsafe"
+
+	"github.com/golang/protobuf/proto"
+	tf "github.com/tensorflow/tensorflow/tensorflow/go"
+)
+
+type Session struct {
+	session tf.TF_Session
+}
+
+func NewSession() (*Session, error) {
+	status := tf.TF_NewStatus()
+	return &Session{
+		session: tf.TF_NewSession(
+			tf.TF_NewSessionOptions(),
+			status,
+		),
+	}, statusToError(status)
+}
+
+type Tensor struct {
+	tensor tf.TF_Tensor
+	buf    []byte
+}
+
+type TensorShape [][]int64
+
+var (
+	TensorShapeScalar = TensorShape{{1}}
+)
+
+func NewTensor(dataType tf.DataType, shape TensorShape, data interface{}) (*Tensor, error) {
+	// TODO(tmc): ensure data is a slice
+	v := reflect.ValueOf(data)
+	if v.Kind() != reflect.Slice {
+		return nil, fmt.Errorf("tensorflow: 'data' argument must be a slice")
+	}
+	dataSize := v.Len() * int(v.Type().Elem().Size())
```

Comment:
I'm perplexed why this is getting cast to an int when it's going to be up-casted to an int64 when used.  (In general, I suspect go+tensorflow should just be all int64 for the sizes of things, unless there's a critical speed optimization I'm overlooking.)